contracts/contracts/PreConfCommitmentStore.sol

```diff
@@ -308,38 +332,45 @@ contract PreConfCommitmentStore is OwnableUpgradeable, UUPSUpgradeable {
 
     /**
      * @dev Verifies a pre-confirmation commitment by computing the hash and recovering the committer's address.
-     * @param txnHash The transaction hash associated with the commitment.
-     * @param bid The bid amount.
-     * @param blockNumber The block number at the time of the bid.
-     * @param bidHash The hash of the bid details.
-     * @param bidSignature The signature of the bid.
-     * @param commitmentSignature The signature of the commitment.
+     * @param params The commitment params associated with the commitment.
      * @return preConfHash The hash of the pre-confirmation commitment.
      * @return commiterAddress The address of the committer recovered from the commitment signature.
      */
+
     function verifyPreConfCommitment(
-        string memory txnHash,
-        uint256 bid,
-        uint64 blockNumber,
-        uint64 decayStartTimeStamp,
-        uint64 decayEndTimeStamp,
-        bytes32 bidHash,
-        bytes memory bidSignature,
-        bytes memory commitmentSignature,
-        bytes memory sharedSecretKey
+        CommitmentParams memory params
     ) public view returns (bytes32 preConfHash, address commiterAddress) {
-        preConfHash = getPreConfHash(
-            txnHash,
-            bid,
-            blockNumber,
-            decayStartTimeStamp,
-            decayEndTimeStamp,
-            bidHash,
-            _bytesToHexString(bidSignature),
-            _bytesToHexString(sharedSecretKey)
+        return _verifyPreConfCommitment(params);
+    }
+
+    function _verifyPreConfCommitment(
+        CommitmentParams memory params
+    ) internal view returns (bytes32 preConfHash, address commiterAddress) {
+        preConfHash = _getPreConfHash(params);
+        commiterAddress = _recoverCommitterAddress(preConfHash, params.commitmentSignature);
+    }
+
+    function _getPreConfHash(
```

Comment:
The public function `getPreConfHash` should instead call the internal `_getPreConfHash` function, where underlying functionality lives in the internal func

Reply:
I think it's fine to keep both, the reason _getPreConfHash was added iirc, is because the yul compiler couldnt handle us passing in varaibles one by one on the stack. Not sure the rationale here, Kant helped fix this one, and I don't want to mess with it too much in fear of reintroducing the same issue.

Reply:
No logic changes are needed, I'm just saying that common solidity pattern is that public functions call internal functions which hold the underlying logic. See `ValidatorRegistryV1.sol` as example

Reply:
This has to do with different gas costs for public/private/internal/external

Reply:
Will make this change later. I haven't stress tested the changed approach.